tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/BaseLinker.scala

```diff
@@ -87,11 +124,20 @@ final class BaseLinker(semantics: Semantics, esLevel: ESLevel, considerPositions
           allowAddingSyntheticMethods = true)
     }
 
-    val linkingErrLevel = if (bypassLinkingErrors) Level.Warn else Level.Error
-    analysis.errors.foreach(logError(_, logger, linkingErrLevel))
+    if (analysis.errors.nonEmpty) {
+      // TODO Make it always fatal when we can get rid of bypassLinkingErrors
+      val fatal = !bypassLinkingErrors || analysis.errors.exists {
+        case _: Analysis.MissingJavaLangObjectClass => true
+        case _: Analysis.CycleInInheritanceChain    => true
+        case _                                      => false
+      }
 
-    if (analysis.errors.nonEmpty && !bypassLinkingErrors)
-      sys.error("There were linking errors")
+      val linkingErrLevel = if (fatal) Level.Error else Level.Warn
+      analysis.errors.foreach(logError(_, logger, linkingErrLevel))
```

Comment:
It's arguable. 

But anyway, in practice it wouldn't change anything, because either you get fatal errors, or you get non fatal errors. There is no way you can get a mix of both.